File: boosting/src/data/dataholder/PredictionData.java
package data.dataholder;

import data.RegressionData;

public class PredictionData implements RegressionData {

    private final double[] window;
    private final double expectedResult;

    public PredictionData(double[] window, double expectedResult) {
        this.window = window;
        this.expectedResult = expectedResult;
    }

    @Override
    public double output() {
        return expectedResult;
    }

    @Override
    public double[] asVector() {
        return window;
    }
}

File: boosting/src/data/dataholder/PredictionData.kt
package data.dataholder

import data.RegressionData

class PredictionData(private val window: DoubleArray, private val expectedResult: Double) : RegressionData {

    override fun output(): Double {
        return expectedResult
    }

    override fun asVector(): DoubleArray {
        return window
    }
}
